### ISSUE TITLE
How to solve question steps by steps using mathsteps

### ISSUE THREAD

**Issue by OmarTawashi** [socraticorg/mathsteps#157]
Welcome 
How to solve question steps by steps using mathsteps
whats is the steps to enter parameter in classes of this library please is necessary ??

![fgstfsdrfsd](https://cloud.githubusercontent.com/assets/9048325/25128370/31a85450-2442-11e7-958c-a3a1f40f7718.png)

**Comment by nitin42** [comment]
Can you elaborate it a bit further ?

**Comment by evykassirer** [comment]
Hi @OmarTawashi I'm not sure what you're asking - could you rephrase?

The screenshot there is from mathpapa and not from our own library

**Comment by salumsinare** [comment]
can i use mathsteps to solve polynomials and calculus problems if true how

**Comment by aliang8** [comment]
Hi @salumsinare,
Glad you asked! Currently mathsteps only supports arithmetic and basic algebra. This applies to polynomials also. You can also factor and simplify polynomials, but that's about as far as it gets. However, we are working towards implementing more complex operations and calculus is definitely on our todo list. Make sure to join the mathsteps gitter discussion to learn more about our progress!

**Comment by evykassirer** [comment]
no to calculus, yes (most of the time) to polynomials

if you give me an example of a problem? and I'll let you know how you can use mathsteps with it

**Comment by salumsinare** [comment]
@evykassirer  
Here is the problem i tried to solve
x^2+3x+2=0 
so far it gives me x^2+3x=-2

**Comment by evykassirer** [comment]
ah! yes that requires factoring - factoring quadratics is actually all coded up and ready to go but not merged or deployed yet. 

Do you need it soon for something or were you just playing around with it? If you'd like to be able to use that functionality soon, I can prioritize getting it ready :)

**Comment by kevinbarabash** [comment]
@salumsinare it looks like only certain forms of quadratics are handled at the moment, see https://github.com/socraticorg/mathsteps/blob/master/lib/ChangeTypes.js#L179-L183.

**Comment by evykassirer** [comment]
#148 added some coverage on top of that too, though didn't actually add anymore changetypes

**Comment by salumsinare** [comment]
@evykassirer  i don't need it that soon but i was just working on a project that requires such functionalities plus calculus

**Comment by salumsinare** [comment]
is there a way to detect the nature of an expression/equation based on mathematical topic or concept and the general method/formula to solve it (if it exist) before actually solving it 
if i enter to solve  x+2=0 it gives something like **linear equation**
and if i do   2x+3<0 it gives inequality

**Comment by aliang8** [comment]
Yup, all we would have to do is scan the expression for those specific relation nodes (<, =, >). From that we can determine whether it is an expression, equation, inequality, etc.

**Comment by salumsinare** [comment]
can mathsteps generate graphs based on an expression or there are any other alternative to achieve this

**Comment by evykassirer** [comment]
That's outside the scope of mathsteps but you can search the web for graphing libraries - a bunch exist

**Comment by nitin42** [comment]
I think we can close this! (outside the scope of project)
